### PR TITLE
Add LemMinX Extension docs

### PR DESCRIPTION
**Pull request by rzgry**
* Adds docs on writing lemminx extensions
* Moves the contents of the wiki to a `/docs` directory to make it easier for external contributors to collaborate.

### CODE REVIEW

File: docs/LemMinX-Extensions.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+- [LemMinX Maven extension](https://github.com/eclipse/lemminx-maven/) provides extra assistance for the Maven pom.xml files.
+- [LemMinX Liquibase extension](https://github.com/Treehopper/liquibase-lsp/) provides extra assistance for Liquibase XML migration scripts, adding database validation (as diagnostics) using an in-memory database.
+- [LemMinX Liberty extension](https://github.com/OpenLiberty/liberty-language-server/tree/master/lemminx-liberty) provides extra assistance for the Open Liberty runtime server.xml file. Adding completion options and diagnostics for Liberty features.
+
+## Creating a LemMinX extension.
+
+LemMinX is extended using the [Java Service Provider Interface (SPI)](https://www.baeldung.com/java-spi). You can extend LemMinX to provide custom completion, hover, diagnostics, renaming etc.
+You can find the complete [LemMinX extension API here](https://github.com/eclipse/lemminx/tree/master/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/extensions).
+
+To start your lemminx extension create a new Java Project that includes LemMinX as a provided dependency. Also make sure to include the `lemminx-releases` repository.
```
Comment (datho7561) [inline docs/LemMinX-Extensions.md:22]: To start a LemMinX extension, ...

File: docs/LemMinX-Extensions.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+## Adding language features
+
+The [LemMinx Extensions API](https://github.com/eclipse/lemminx/tree/master/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/extensions) supports many of the language features defined in the [Language server Protocol](https://microsoft.github.io/language-server-protocol/). You can add these features to your extension by implementing the feature participant and registering the participant in your implementation of IXMLExtension. This includes:
+
+- Code actions with `ICodeActionParticipant`
+- Code Completion with `ICompletionParticipant`
+- Go to Definition with `IDefinitionParticipant`
+- Finding the location of document Links with `IDocumentLinkParticipant`
```
Comment (datho7561) [inline docs/LemMinX-Extensions.md:59]: maybe "adding document links" would be a bit more clear?

File: docs/LemMinX-Extensions.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+## Testing your custom LemMinX extension
+
+If you are using lemminx version >= `0.14.0` to build your extension you can use the lemminx [XMLAssert](https://github.com/eclipse/lemminx/blob/master/org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/XMLAssert.java) class to test the functionality of your extension.
+
+XMLAssert provides many static helper methods for building lsp4j structures such as CompletionItem, TextEdit, Hover etc. 
+
+For example to build a CompletionItem that uses a TextEdit you can use the `c` and `te` methods.
+```
```
Comment (datho7561) [inline docs/LemMinX-Extensions.md:77]: use \`\`\`java in order to get java syntax highlighting

File: docs/LemMinX-Extensions.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+CompletionItem testCompletionItem = c("xsl:template", te(2, 0, 2, 0, "<xsl:template></xsl:template>"), "xsl:template");
+```
+
+Along with building lsp4j structures, XMLAssert contains static methods that test xml language functionality.
+
+For example to test that completion options exist, we build an XML String that contains a `|` character. The `|` is used to represent where the completion has been triggered in the XML file.
+
+```
```
Comment (datho7561) [inline docs/LemMinX-Extensions.md:85]: use \`\`\`java in order to get java syntax highlighting

**Review by datho7561 — CHANGES_REQUESTED** [review]

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/settings/XMLGeneralClientSettings.java [hunk 5]
```diff
@@ -17,7 +17,7 @@
 /**
  * Class to hold all settings from the client side.
  * 
- * See https://github.com/eclipse/lemming/wiki/Configuration for more
+ * See https://github.com/rzgry/lemminx/blob/lemminx-ext-docs/docs/Configuration.md for more
```
Comment (datho7561) [inline org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/settings/XMLGeneralClientSettings.java:20]: Change to `https://github.com/eclipse/lemminx/blob/master/docs/Configuration.md`

Comment (rzgry) [reply org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/settings/XMLGeneralClientSettings.java]: Whoops 😄 good catch fixed.

**Review by datho7561 — APPROVED** [review]

**Pull request merged**